charts/vsphere-csi-secret/templates/vsphere-csi-secret.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: vsphere-config-secret
+  namespace: kube-system
+spec:
+  refreshInterval: 5m
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: {{ .Values.secretStore }}
+  data:
+    - remoteRef:
+        key: {{ .Values.vaultPath }}
+        property: password
+      secretKey: password
+    - remoteRef:
+        key: {{ .Values.vaultPath }}
+        property: username
+      secretKey: username
+  target:
+    name: vsphere-config-secret
+    template:
+      engineVersion: v2
+      data:
+        csi-vsphere.conf: |
+          [Global]
+          cluster-id = "{{ .Values.clusterName }}-cluster"
```

Comment:
@DanielStano , pls could you verify cluster-id 

According to https://github.com/gr8it/conf-lab.gr8it.cloud/tree/main/fleet-applications-common/dev/vsphere-csi

cluster-id = "customer_environment_clustername" 

Thx
P.

Reply:
so far the customer_environment_clustername is not yet implemented in downstream dev or stable, so i can only work with clusterName as in the chart. an unique cluster-id is needed but so far not implemented. i can either wait and put the work on hold or rework the chart as needed once labeling is implemented in our env

Reply:
I believe in power of  helm templating, with Haimdall help you can do it.
https://helm.sh/docs/chart_template_guide/control_structures/

Reply:
changed to cluster-id = "{{ .Values.customer }}_{{ .Values.environment }}_{{ .Values.clusterName }}" as per yourself